app/services/global_search/statement_searcher.rb

```diff
@@ -0,0 +1,38 @@
+module GlobalSearch
+
+  class StatementSearcher
+
+    def initialize(user = nil)
+      @user = user
+    end
+
+    def search(query)
+      query = query.to_s
+                   .strip # get rid of leading/trailing whitespace
+                   .sub(/\A#/, "") # remove a leading hash sign to support searching like "#123-456"
+
+      statement = Statement.find_by_invoice_number(query)
```

Comment:
This one actually isn't one of the dynamic finders. It's a method I created. But the other one makes sense to add a TODO.